<?xml version="1.0"?>
<robot xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
       xmlns:xacro="http://ros.org/wiki/xacro">

	<xacro:property name="wheel_mechanical_reduction" value="1" />

	<xacro:macro name="youbot_base_transmission">
	<!-- front left wheel -->
		<transmission name="wheel_trans_fl">
      <type>transmission_interface/SimpleTransmission</type>
			<actuator name="wheel_motor_fl">
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        <mechanicalReduction>${wheel_mechanical_reduction}</mechanicalReduction>
      </actuator>
			<joint name="wheel_joint_fl">
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      </joint>
		</transmission>

		
	  <!-- front right wheel -->
	  <transmission name="wheel_trans_fr">
      <type>transmission_interface/SimpleTransmission</type>
		  <actuator name="wheel_motor_fr">
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        <mechanicalReduction>${wheel_mechanical_reduction}</mechanicalReduction>
      </actuator>
		  <joint name="wheel_joint_fr">
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      </joint>
	  </transmission>

	 

	  <!-- back left wheel -->
	  <transmission name="wheel_trans_bl">
      <type>transmission_interface/SimpleTransmission</type>
	    <actuator name="wheel_motor_bl" >
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        <mechanicalReduction>${wheel_mechanical_reduction}</mechanicalReduction>
      </actuator>
	    <joint name="wheel_joint_bl">
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      </joint>
	  </transmission>

	  

	  <!-- back right wheel -->
	  <transmission name="wheel_trans_br">
      <type>transmission_interface/SimpleTransmission</type>
	    <actuator name="wheel_motor_br" >
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        <mechanicalReduction>${wheel_mechanical_reduction}</mechanicalReduction>
      </actuator>
		  <joint name="wheel_joint_br">
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      </joint>
	  </transmission>

	  

  </xacro:macro>
	
</robot>
